bentoml/pytorch.py

```diff
@@ -230,28 +232,33 @@ def _setup(self) -> None:  # type: ignore[override]
     @torch.no_grad()
     def _run_batch(  # type: ignore[override]
         self,
-        *args: t.Union[np.ndarray, "torch.Tensor"],
+        *args: t.Union[np.ndarray, torch.Tensor],
         **kwargs: str,
-    ) -> "torch.Tensor":
+    ) -> torch.Tensor:
 
-        # We assume *args are input data. They share the same type
-        # (tensor or ndarray) while **kwargs are configuration
-        # parameters, which may have type str, int, float etc.
+        params = Params[t.Any](*args, **kwargs)
 
-        params = Params[t.Union[np.ndarray, "torch.Tensor"]](*args)
-        sample = params.sample
+        def _mapping(item) -> t.Any:
+            if isinstance(item, np.ndarray):
+                item = torch.from_numpy(item)
 
-        if isinstance(sample, np.ndarray):
-            params = params.map(lambda i: torch.from_numpy(i))
-        if self.resource_quota.on_gpu:
-            params = params.map(lambda i: i.cuda())
+            if self.resource_quota.on_gpu:
+                if isinstance(item, torch.Tensor):
+                    item = item.cuda()
 
-        res: "torch.Tensor"
+            return item
+
+        params = params.map(_mapping)
+
+        # merge kwargs with partial_kwargs
+        merged_kwargs = {**self._partial_kwargs, **kwargs}
```

Comment:
any updates from our previous chat? @larme

Reply:
@aarnphm done, just wanted to finish the test then do this so the refactoring is tested~

Reply:
i saw the push, well done 👍🏼